packages/contracts/tasks/deposits.ts

```diff
@@ -52,19 +70,45 @@ task('deposit', 'Deposits funds onto L2.')
     ).connect(l1Wallet)
 
     const amountWei = utils.parseEther(amountEth)
-    console.log(`Depositing ${amountEth} ETH to ${to}...`)
+    const value = amountWei.add(utils.parseEther('0.01'))
+    console.log(`Depositing ${amountEth} ETH to ${to}`)
     // Below adds 0.01 ETH to account for gas.
     const tx = await depositFeed.depositTransaction(
       to,
       amountWei,
       '3000000',
       false,
       [],
-      {
-        value: amountWei.add(utils.parseEther('0.01')),
-      }
+      { value }
     )
     console.log(`Got TX hash ${tx.hash}. Waiting...`)
-    await tx.wait()
-    console.log('Done.')
+    const receipt = await tx.wait()
+
+    const event = receipt.events[0]
+    if (event?.event !== 'TransactionDeposited') {
+      throw new Error('Transaction not deposited')
+    }
+
```

Comment:
I think the best devex from a user perspective would be something like:

```js
const l2Tx = DepositTx.fromReceipt(receipt)
```
Then it could handle the boilerplate

Reply:
Maybe call it `fromL1Receipt` to avoid confusion, since deposits have a receipt on both L1 and L2